katsdpimager/frontend.py

```diff
@@ -192,7 +192,10 @@ def find_peak(image, pbeam, noise):
 @profile_function()
 def get_totals(image_parameters, image, restoring_beam):
     """Compute total flux density in each polarization."""
-    sums = np.nansum(image, axis=(1, 2), dtype=np.float64)   # Sum separately per polarization
+    # Using `where` is a faster version of np.nansum
+    # (see https://github.com/numpy/numpy/issues/12662).
```

Comment:
Reading the issue turned me onto https://pypi.org/project/Bottleneck/...

A nice option for things like `nanmedian` (5x faster).

Reply:
I also saw that and gave it a try, but for nansum it's no faster than this approach, and it didn't seem worth adding a dependency as it's not pure Python.